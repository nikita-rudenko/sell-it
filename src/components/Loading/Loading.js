import React from 'react';

import CSSModules from 'react-css-modules';
import styles from './Loading.module.scss';
import loading from 'assets/img/gif/loading.gif';

const Loading = () => {
  return (
    <div styleName='container'>
      <img src={loading} alt='Loading...' />
      <div styleName='loading'>Loading</div>
    </div>
  );
};

export default CSSModules(Loading, styles);
