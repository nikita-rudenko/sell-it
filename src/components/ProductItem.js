import React, { Component } from 'react';

export default class ProductItem extends Component {
	render() {
		return (
			<div className="product">
				<div className="product__image-wrapper">
					<img
						className="product__image"
						src="img/product-min.png"
						alt="Product"
					/>
				</div>
				<div className="product__info">
					<span className="product__title">Product Title</span>
					<span className="product__btn">
						<img src="img/icons/fa-eye.png" alt="Eye" />
					</span>
				</div>
			</div>
		);
	}
}
