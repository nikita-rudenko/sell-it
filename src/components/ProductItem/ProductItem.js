import React, { Component } from 'react';
import styles from './ProductItem.module.scss';
import CSSModules from 'react-css-modules';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { setProductID } from '../../actions/ProductActions';
import defImg from '../../img/default.jpg';
import eye from '../../img/icons/fa-eye.png';

class ProductItem extends Component {
  setProductDetailsID(id) {
    this.props.setProductID(id);
  }

  render() {
    const { theme, images, pk } = this.props.item;

    const image = images.length ? images[0].file : defImg;
    const title = theme ? theme : 'Product Title';

    return (
      <Link
        styleName='product'
        to='/details'
        onClick={() => this.setProductDetailsID(pk)}
      >
        <div styleName='image-wrapper'>
          <img styleName='image' src={image} alt='Product' />
        </div>
        <div styleName='info'>
          <span styleName='title'>{title}</span>
          <img src={eye} alt='Eye' />
        </div>
      </Link>
    );
  }
}

const mapStateToProps = state => ({
  details: state.details
});

const mapDispatchToProps = dispatch => ({
  setProductID: id => dispatch(setProductID(id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CSSModules(ProductItem, styles));
