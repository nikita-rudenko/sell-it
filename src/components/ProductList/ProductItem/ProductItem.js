import React from 'react';
import styles from './ProductItem.module.scss';

export default function ProductItem() {
	return (
		<div className={styles.product}>
			<div className={styles.imageWrapper}>
				<img className={styles.image} src="img/product-min.png" alt="Product" />
			</div>
			<div className={styles.info}>
				<span className={styles.title}>Product Title</span>
				<span className={styles.btn}>
					<img src="img/icons/fa-eye.png" alt="Eye" />
				</span>
			</div>
		</div>
	);
}
