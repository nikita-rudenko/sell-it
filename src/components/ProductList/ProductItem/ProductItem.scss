@import '../../../sass/variables.scss';

.product {
	height: 25rem;
	flex: 0 31%;
	margin: 1rem;
	box-shadow: $shadow-dark;
	font-family: 'Arial-BoldMT';

	@media only screen and (max-width: $bp-medium) {
		flex: 0 46.5%;
	}

	@media only screen and (max-width: $bp-small) {
		flex: 0 100%;
	}

	&__image-wrapper {
		width: 100%;
		height: 80%;
	}

	&__image {
		width: 100%;
		height: 100%;
		object-fit: cover;
	}

	&__info {
		background-color: $color-primary;
		color: $color-white;
		height: 5rem;
		display: flex;
		justify-content: space-between;
		align-items: center;
		padding: 0 1rem;
	}

	&__title {
		font-size: 1.6rem;
	}
}
