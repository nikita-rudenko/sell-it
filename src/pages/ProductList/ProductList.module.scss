.product-list {
  max-width: 1200px;
  margin: 0 auto 0 auto;
  padding: 2rem 4.5rem;
  display: flex;
  flex-flow: row wrap;

  @media only screen and (max-width: 56.25em) {
    padding: 0 3rem;
  }

  @media only screen and (max-width: 37.5em) {
    justify-content: center;
  }
}

.no-results {
  width: 100%;
  text-align: center;
  font-size: 1.6rem;
}

.pagination {
  width: 100%;
  margin: 5rem 5rem 10rem 5rem;
}
