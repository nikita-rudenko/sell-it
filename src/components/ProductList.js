import React from 'react';
import Header from './Header';
import ProductItem from './ProductItem';
import Footer from './Footer';

export default function ProductList(props) {
	return (
		<>
			<Header shift={props.shift} />
			<main>
				<section className="product-list u-mt-huge">
					<ProductItem />
					<ProductItem />
					<ProductItem />
					<ProductItem />
					<ProductItem />
					<ProductItem />
					<ProductItem />
					<ProductItem />
					<ProductItem />
					<ProductItem />
					<ProductItem />
				</section>
			</main>
			<Footer />
		</>
	);
}
