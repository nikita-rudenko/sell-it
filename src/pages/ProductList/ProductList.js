import React, { Component } from 'react';
import styles from './ProductList.module.scss';
import CSSModules from 'react-css-modules';
import { connect } from 'react-redux';
import { fetchProducts } from '../../actions/productActions';
import Header from '../../components/Header/Header';
import ProductItem from '../../components/ProductItem/ProductItem';
import Footer from '../../components/Footer/Footer';
import Loading from '../../components/Loading/Loading';

class ProductList extends Component {
  componentDidMount() {
    this.props.fetchProducts();
  }

  mapProducts = data => {
    return data ? (
      data.map(item => {
        return <ProductItem key={item.pk} item={item} />;
      })
    ) : (
      <div>"An error occured"</div>
    );
  };

  render() {
    const { productList, isFetching } = this.props;
    const data = productList.data;

    return (
      <>
        <Header />
        <main>
          {isFetching ? (
            <Loading />
          ) : (
            <section styleName='product-list'>{this.mapProducts(data)}</section>
          )}
        </main>
        <Footer />
      </>
    );
  }
}

const mapStateToProps = state => ({
  productList: state.product.productList,
  isFetching: state.product.isFetching
});

const mapDispatchToProps = dispatch => ({
  fetchProducts: () => dispatch(fetchProducts())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CSSModules(ProductList, styles));
