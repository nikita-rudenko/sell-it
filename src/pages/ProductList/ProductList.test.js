import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { ProductList } from './ProductList';
import Loading from 'components/Loading/Loading';
import ProductItem from 'components/ProductItem/ProductItem';

Enzyme.configure({ adapter: new Adapter() });

describe('ProductList tests', () => {
  const props = {
    isFetching: false,
    data: null
  };

  describe('ProductList initial', () => {
    const productList = shallow(<ProductList {...props} />, {
      disableLifecycleMethods: true
    });

    it("doesn't render ProductItems", () => {
      expect(productList.find(ProductItem)).toHaveLength(0);
    });

    it("doesn't render <Loading />", () => {
      expect(productList.find(Loading)).toHaveLength(0);
    });

    it('renders properly', () => {
      expect(productList).toMatchSnapshot();
    });
  });

  describe('ProductList is fetching', () => {
    const nextProps = {
      ...props,
      isFetching: true
    };

    const productList = shallow(<ProductList {...nextProps} />, {
      disableLifecycleMethods: true
    });

    it('renders preloader', () => {
      expect(productList.find(Loading)).toHaveLength(1);
    });

    it("doesn't render <p>", () => {
      expect(productList.find('p')).toHaveLength(0);
    });

    it("doesn't render <ProductList />", () => {
      expect(productList.find(ProductItem)).toHaveLength(0);
    });

    it('renders properly', () => {
      expect(productList).toMatchSnapshot();
    });
  });

  describe('ProductList renders items', () => {
    const nextProps = {
      isFetching: false,
      productList: [
        {
          pk: 344,
          theme: 'Telefon',
          text: 'Testovy text'
        }
      ]
    };

    const productList = shallow(<ProductList {...nextProps} />, {
      disableLifecycleMethods: true
    });

    it('renders items', () => {
      expect(productList.find(ProductItem)).toHaveLength(1);
    });

    it("doesn't render preloader", () => {
      expect(productList.find(Loading)).toHaveLength(0);
    });

    it("doesn't render <p>", () => {
      expect(productList.find('p')).toHaveLength(0);
    });

    it('renders properly', () => {
      expect(productList).toMatchSnapshot();
    });
  });

  describe('ProductList no data received', () => {
    const nextProps = {
      isFetching: false,
      productList: []
    };

    const productList = shallow(<ProductList {...nextProps} />, {
      disableLifecycleMethods: true
    });

    it('renders <p>', () => {
      expect(productList.find('p')).toHaveLength(0);
    });

    it("doesn't render items", () => {
      expect(productList.find(ProductItem)).toHaveLength(0);
    });

    it("doesn't render preloader", () => {
      expect(productList.find(Loading)).toHaveLength(0);
    });

    it('renders properly', () => {
      expect(productList).toMatchSnapshot();
    });
  });
});
