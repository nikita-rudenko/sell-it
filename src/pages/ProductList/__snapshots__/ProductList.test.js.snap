// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList tests ProductList initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductList
    data={null}
    isFetching={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <p>
        No data...
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No data...",
      },
      "ref": null,
      "rendered": "No data...",
      "type": "p",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <p>
          No data...
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No data...",
        },
        "ref": null,
        "rendered": "No data...",
        "type": "p",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`ProductList tests ProductList is fetching renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductList
    data={null}
    isFetching={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Loading />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Loading />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`ProductList tests ProductList no data received renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductList
    isFetching={false}
    productList={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <section
        styleName="product-list"
      >
        <div
          styleName="no-results"
        >
          <h1>
            No results
          </h1>
        </div>
      </section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          styleName="no-results"
        >
          <h1>
            No results
          </h1>
        </div>,
        "styleName": "product-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            No results
          </h1>,
          "styleName": "no-results",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No results",
          },
          "ref": null,
          "rendered": "No results",
          "type": "h1",
        },
        "type": "div",
      },
      "type": "section",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <section
          styleName="product-list"
        >
          <div
            styleName="no-results"
          >
            <h1>
              No results
            </h1>
          </div>
        </section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            styleName="no-results"
          >
            <h1>
              No results
            </h1>
          </div>,
          "styleName": "product-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              No results
            </h1>,
            "styleName": "no-results",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No results",
            },
            "ref": null,
            "rendered": "No results",
            "type": "h1",
          },
          "type": "div",
        },
        "type": "section",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`ProductList tests ProductList renders items renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductList
    isFetching={false}
    productList={
      Array [
        Object {
          "pk": 344,
          "text": "Testovy text",
          "theme": "Telefon",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <section
        styleName="product-list"
      >
        <ProductItem
          item={
            Object {
              "pk": 344,
              "text": "Testovy text",
              "theme": "Telefon",
            }
          }
        />
      </section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ProductItem
            item={
              Object {
                "pk": 344,
                "text": "Testovy text",
                "theme": "Telefon",
              }
            }
          />,
        ],
        "styleName": "product-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "344",
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "pk": 344,
              "text": "Testovy text",
              "theme": "Telefon",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <section
          styleName="product-list"
        >
          <ProductItem
            item={
              Object {
                "pk": 344,
                "text": "Testovy text",
                "theme": "Telefon",
              }
            }
          />
        </section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ProductItem
              item={
                Object {
                  "pk": 344,
                  "text": "Testovy text",
                  "theme": "Telefon",
                }
              }
            />,
          ],
          "styleName": "product-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "344",
            "nodeType": "function",
            "props": Object {
              "item": Object {
                "pk": 344,
                "text": "Testovy text",
                "theme": "Telefon",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
